core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/CASLockFactory.java

```diff
@@ -199,17 +206,40 @@ private CASLockFactory(final Builder builder)
 
         myUuid = hostId;
 
-        myLockCache = new LockCache(this::doTryLock);
+        int lockTimeInSeconds = getDefaultTimeToLiveFromLockTable();
+        int lockUpdateTimeInSeconds = lockTimeInSeconds / REFRESH_INTERVAL_RATIO;
+        int myFailedLockRetryAttempts = (lockTimeInSeconds / lockUpdateTimeInSeconds) - 1;
+        myCasLockFactoryCacheContext = CASLockFactoryCacheContext.newBuilder()
+                .withLockUpdateTimeInSeconds(lockUpdateTimeInSeconds)
+                .withFailedLockRetryAttempts(myFailedLockRetryAttempts)
+                .withLockCache(new LockCache(this::doTryLock, builder.myCacheExpiryTimeInSeconds))
+                .build();
+    }
+
+    private int getDefaultTimeToLiveFromLockTable()
+    {
+        TableMetadata tableMetadata = mySession.getMetadata()
+                .getKeyspace(myKeyspaceName)
+                .flatMap(ks -> ks.getTable(TABLE_LOCK))
+                .orElse(null);
+
```

Comment:
I think this would be more readable.
if (tableMetadata == null || tableMetadata.getOptions() == null)
{
   LOG.warn("Could not parse default ttl of {}.{}", myKeyspaceName. TABLE_LOCK);
   return DEFAULT_LOCK_TIME_IN_SECONDS;
}
Map<CqlIdentifier, Object> tableOptions = tableMetadata.getOptions();
return (Integer) tableOptions.get(CqlIdentifier.fromInternal("default_time_to_live"));